lightning/src/ln/channel.rs

```diff
@@ -2953,6 +2953,28 @@ impl<Signer: Sign> Channel<Signer> {
 			panic!("Cannot update fee while peer is disconnected/we're awaiting a monitor update (ChannelManager should have caught this)");
 		}
 
+		// Before proposing a feerate update, check that we can actually afford the new fee.
+		let inbound_stats = self.get_inbound_pending_htlc_stats(Some(feerate_per_kw));
+		let outbound_stats = self.get_outbound_pending_htlc_stats(Some(feerate_per_kw));
+		let total_fee = feerate_per_kw as u64 * (COMMITMENT_TX_BASE_WEIGHT + (inbound_stats.pending_htlcs as u64 + outbound_stats.pending_htlcs as u64) * COMMITMENT_TX_WEIGHT_PER_HTLC) / 1000;
```

Comment:
suggestion: This computation is spelled out like 10x across this file, would ease review burden if there were a utility for it

Reply:
See new `commit_tx_fee_sat` :) Though lmk if you have a suggestion to avoid confusion with `commit_tx_fee_msat` beyond doc.